### PR TITLE
Soc nrf radio coded phy capabilities

### PR DESCRIPTION
**Pull request by ioannisg**
A minor patch that effectively allows the nRF LE Controller to build with Coded PHY support for SoCs other than nRF52840. 

No functional changes in this PR.

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Comment by ioannisg** [comment]
BTW, this allows to use LE Coded PHY on nRF52811, so that's actually, also a bug fix patch

**Comment by ioannisg** [comment]
@anangl pls, have a quick look

File: soc/arm/nordic_nrf/nrf52/Kconfig.soc [hunk 1]
```diff
@@ -64,6 +64,7 @@ config SOC_NRF52811
 	select HAS_HW_NRF_PWM0
 	select HAS_HW_NRF_QDEC
 	select HAS_HW_NRF_RADIO_IEEE802154
+	select HAS_HW_NRF_RADIO_BLE_CODED
```
Comment (anangl) [inline soc/arm/nordic_nrf/nrf52/Kconfig.soc]: Keep the alphabetical order, please. In `Kconfig.peripheral` as well.

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -406,7 +406,7 @@ config BT_CTLR_PHY_2M
 
 config BT_CTLR_PHY_CODED
 	bool "Coded PHY Support"
-	depends on (BT_CTLR_PHY || BT_CTLR_ADV_EXT) && SOC_NRF52840
+	depends on (BT_CTLR_PHY || BT_CTLR_ADV_EXT) && HAS_HW_NRF_RADIO_BLE_CODED
```
Comment (anangl) [inline subsys/bluetooth/controller/Kconfig]: Isn't the line too long now?

**Review by anangl — APPROVED** [review]
Overall looks good. Minor corrections suggested only.

File: subsys/bluetooth/controller/Kconfig (hunk 2 above)
Comment (ioannisg) [reply subsys/bluetooth/controller/Kconfig]: Yes, but this is common in Kconfig files, AFAIK

File: soc/arm/nordic_nrf/nrf52/Kconfig.soc (hunk 1 above)
Comment (ioannisg) [reply soc/arm/nordic_nrf/nrf52/Kconfig.soc]: Right. Will do that, thanks.

Comment (ioannisg) [reply soc/arm/nordic_nrf/nrf52/Kconfig.soc]: Fixed

**Review by cvinayak — APPROVED** [review]
LGTM

**Pull request merged**